src/test/app/Escrow_test.cpp

```diff
@@ -1656,6 +1656,361 @@ struct Escrow_test : public beast::unit_test::suite
+                finish_time(env.now() + 2s),
+                fee(txnFees));
+            env.close();
+        }
+        {
+            // FinishFunction + FinishAfter + Condition + CancelAfter
+            env(escrowCreate,
+                finish_function(wasmHex),
+                cancel_time(env.now() + 100s),
+                condition(cb1),
+                finish_time(env.now() + 2s),
+                fee(txnFees));
+            env.close();
+        }
+
+        // Failure situations (i.e. all other combinations)
+        {
+            // only FinishFunction
+            env(escrowCreate,
+                finish_function(wasmHex),
+                fee(txnFees),
+                ter(temBAD_EXPIRATION));
+            env.close();
+        }
+        {
+            // FinishFunction + FinishAfter
+            env(escrowCreate,
+                finish_function(wasmHex),
+                finish_time(env.now() + 2s),
+                fee(txnFees),
+                ter(temBAD_EXPIRATION));
+            env.close();
+        }
+        {
+            // FinishFunction + Condition
+            env(escrowCreate,
+                finish_function(wasmHex),
+                condition(cb1),
+                fee(txnFees),
+                ter(temBAD_EXPIRATION));
+            env.close();
+        }
+        {
+            // FinishFunction + FinishAfter + Condition
+            env(escrowCreate,
+                finish_function(wasmHex),
+                condition(cb1),
+                finish_time(env.now() + 2s),
+                fee(txnFees),
+                ter(temBAD_EXPIRATION));
+            env.close();
+        }
+        {
+            // FinishFunction 0 length
+            env(escrowCreate,
+                finish_function(""),
+                cancel_time(env.now() + 100s),
+                fee(txnFees),
+                ter(temMALFORMED));
+            env.close();
+        }
+        // {
+        //     // FinishFunction > max length
+        //     std::string longWasmHex = "00";
+        //     // TODO: fix to use the config setting
+        //     // TODO: make this test more efficient
+        //     // uncomment when that's done
+        //     for (int i = 0; i < 0xffffffff; i++)
+        //     {
+        //         longWasmHex += "11";
+        //     }
+        //     env(escrowCreate,
+        //         finish_function(longWasmHex),
+        //         cancel_time(env.now() + 100s),
+        //         fee(txnFees),
+        //         ter(temMALFORMED));
+        //     env.close();
+        // }
+    }
+
+    void
+    testFinishFunction()
+    {
+        testcase("PoC escrow function");
+
+        using namespace jtx;
+        using namespace std::chrono;
+
+        Account const alice{"alice"};
+        Account const carol{"carol"};
+
+        // Tests whether the ledger index is >= 5
+        static auto wasmHex =
+            "0061736d010000000105016000017f021b0108686f73745f6c69620e6765745f6c"
```

Comment:
Please add pseudo code for this function in comments

Reply:
Fixed in [c57c5a3](https://github.com/XRPLF/rippled/pull/5340/commits/c57c5a31e7376991f9f1613cf99e552442295c89)